### DISCUSSION TITLE
Styling a component is always undefined (e.g. styled(Button))

### DISCUSSION THREAD

**Discussion by bradypp** [gatsbyjs/gatsby#26204] [Help] [answered]
# Summary

I've tried to find answers for a while and I'm surprised that I can't find any.

Whenever I try to style a custom component like a Button or Header using styled-components, it always returns undefined.

I think I must've missed something obvious but don't know what.

## Relevant information

I've installed `gatsby-plugin-styled-components` and have it listed as a plugin in gatsby-config.js.
Each of my custom components has a className prop.
Using styled(props => <Components {...props}/>) works but I need to use !important to override styles.

Thanks for any help.

**Comment by mxstbr** [comment, marked as answer]
Can you share the code that isn't working? This definitely works for me:

```JS
const Button = (props) => <button {...props} />

const StyledButton = styled(Button)`
  color: blue;
`
```

https://codesandbox.io/s/elated-montalcini-0tzqg?file=/src/App.js:0-260

**Comment by bradypp** [comment]
This has always worked for me using CRA but I can't figure out why it isn't working with Gatsby. I'm not sure which code to share as I don't know where the problem is.

If I try to style a Button like you just described, I get an error saying the component Button is undefined.

I'd appreciate it if you have a quick look at the repo: https://github.com/bradypp/personal-website

**Comment by mxstbr** [comment]
I'm not sure I understand, sorry 🤔  Are you seeing an error message pop up? If so, where does that error message point?
